src/tools/rbd_mirror/ImageSyncThrottler.h

```diff
@@ -33,56 +31,49 @@ namespace librbd { namespace journal { struct MirrorPeerClientMeta; } }
 namespace rbd {
 namespace mirror {
 
-template <typename> class ImageSync;
+template <typename> class InstanceWatcher;
 
 class ProgressContext;
 
 /**
  * Manage concurrent image-syncs
  */
 template <typename ImageCtxT = librbd::ImageCtx>
-class ImageSyncThrottler : public md_config_obs_t {
+class ImageSyncThrottler {
```

Comment:
`InstanceSyncThrottler` name was chosen when I was working on previous version, when this class was handling sync notifications via the leader object. Now this name really looks wrong. I wouldn't like to change it to `ImageSyncThrottler` because it knows nothing about 'image', and even that operations it throttles are syncs. I am going to change it to just `Throttler`.

I will evaluate your idea to remove `ImageSyncThrottler` and use `InstanceWatcher::notify_sync_xyz` methods directly in `ImageSync` state machine. Thanks.